apps/public/src/app/(general)/bijlage/Attachments.test.tsx

```diff
@@ -0,0 +1,138 @@
+
+    render(<Attachments {...defaultProps} />)
+
+    const fileInput = screen.getByLabelText(/Selecteer bestanden/i) as HTMLInputElement
+
+    const file = new File(['dummy content'], 'Screenshot 2025-02-10 at 08.29.41.png', { type: 'image/png' })
+
+    await user.upload(fileInput, [file])
+
+    const fileName1 = screen.getByText('Screenshot 2025-02-10 at 08.29.41.png')
+
+    expect(fileName1).toBeInTheDocument()
+
+    const deleteButtons = screen.getAllByRole('button')
+
+    await user.click(deleteButtons[0])
+
+    const file1SecondRender = screen.queryByText('Screenshot 2025-02-10 at 08.29.41.png')
+
+    expect(file1SecondRender).not.toBeInTheDocument()
+  })
+
+  it('should throw an error when delete request fails', async () => {
+    server.use(http.delete(ENDPOINTS.MELDING_ATTACHMENT_DELETE_BY_ID, () => HttpResponse.error()))
+    const user = userEvent.setup()
+
+    render(<Attachments {...defaultProps} />)
+
+    const fileInput = screen.getByLabelText(/Selecteer bestanden/i) as HTMLInputElement
+
+    const file = new File(['dummy content'], 'Screenshot 2025-02-10 at 08.29.41.png', { type: 'image/png' })
+
+    await user.upload(fileInput, [file])
+
+    const fileName1 = screen.getByText('Screenshot 2025-02-10 at 08.29.41.png')
+
+    expect(fileName1).toBeInTheDocument()
+
+    const deleteButtons = screen.getAllByRole('button')
+
+    await user.click(deleteButtons[0])
+
+    const file1SecondRender = screen.queryByText('Screenshot 2025-02-10 at 08.29.41.png')
+
+    expect(file1SecondRender).toBeInTheDocument()
+  })
+
+  it('should show an error when post fails', async () => {
+    server.use(http.post(ENDPOINTS.MELDING_ATTACHMENT_BY_ID, () => HttpResponse.error()))
+
+    const user = userEvent.setup()
+
+    render(<Attachments {...defaultProps} />)
+
+    const fileInput = screen.getByLabelText(/Selecteer bestanden/i) as HTMLInputElement
+
+    const file = new File(['dummy content'], 'Screenshot 2025-02-10 at 08.29.41.png', { type: 'image/png' })
+
+    await user.upload(fileInput, [file])
+
+    const errorMessage = screen.getByText('Failed to fetch')
+
+    expect(errorMessage).toBeInTheDocument()
+  })
+
+  it('should throw an error when attempting to upload too many files', async () => {
+    const user = userEvent.setup()
+
+    render(<Attachments {...defaultProps} />)
+
+    const fileInput = screen.getByLabelText(/Selecteer bestanden/i) as HTMLInputElement
+
+    const file = new File(['dummy content'], 'example.png', { type: 'image/png' })
+    const file2 = new File(['dummy content two'], 'example2.png', { type: 'image/png' })
+    const file3 = new File(['dummy content three'], 'example3.png', { type: 'image/png' })
+    const file4 = new File(['dummy content four'], 'example4.png', { type: 'image/png' })
+
+    await user.upload(fileInput, [file, file2, file3, file4])
+
+    const errorMessage = screen.getByText('U kunt maximaal 3 bestanden uploaden.')
+
+    expect(fileInput.files).toHaveLength(4)
```

Comment:
Is this what we expect? I would assume the file input would be cleared of files on error

Reply:
Might be not the best check indeed, but has no functional impact. I the check for file names.

Reply:
>  no functional impact

I'm not sure about this. Aren't we sending the files to the backend on submit when the `input type="file"` isn't cleared?

Reply:
Resetting the form.